tests/test_api.py

```diff
@@ -0,0 +1,49 @@
+import unittest
```

Comment:
pytest or unittests? IMO pytest won the unittest wars on python, so I would go with pytest

Reply:
OK, I've no particular preferences and it seems pytest runs unittest test out of the box https://docs.pytest.org/en/latest/unittest.html

Reply:
I am also in strong favor for pytest.

Reply:
OK, now it uses `unittest`. This required some ugly changes to the config because Travis pytest behavior [changed slightly between python versions](https://github.com/travis-ci/docs-travis-ci-com/issues/982) but it works.

We are looking forward to try `pipenv` for a saner dependency management.

Reply:
I am also having an eye on pipenv. But currently it is still evolving very quickly.